packages/web3-eth/src/rpc_method_wrappers.ts

```diff
@@ -1068,42 +1085,34 @@ export function sendTransaction<
 		(resolve, reject) => {
 			setImmediate(() => {
 				(async () => {
-					try {
-						let transactionFormatted = formatTransaction(
-							{
-								...transaction,
-								from: getTransactionFromOrToAttr('from', web3Context, transaction),
-								to: getTransactionFromOrToAttr('to', web3Context, transaction),
-							},
-							ETH_DATA_FORMAT,
-						);
-
-						if (web3Context.handleRevert) {
-							// eslint-disable-next-line no-use-before-define
-							await getRevertReason(
```

Comment:
`1.x` doesn't send an `eth_call` before every sent transaction, and `eth_call` will not return an error for transactions such as a send to this method:

```solidity
function revertSend(string memory revertString) public {
    require(keccak256(abi.encodePacked(revertString)) != keccak256(abi.encodePacked("revert")), "This is a send revert");
}
```

so relying on getting the revert reason without sending the transaction isn't a complete solution. Instead, I would suggest checking for a revert reason before sending the transaction only if `options.checkRevertBeforeSend` (a new transaction option) is `true`, and we can set it to default to `true` since this would be helpful to new ETH users and more experienced devs will just turn it off if unwanted

Reply:
There also appears to be a discrepancy between providers with returning revert errors when using `eth_call`:

Sending the transaction:

```typescript
{
    from: tempAcc.address,
    to: '0x0000000000000000000000000000000000000000',
    value: BigInt('999999999999999999999999999999999999999999999999999999999'),
}
```

yields `err: insufficient funds for gas * price + value: address` when submitted to Geth using `eth_call`, but Ganache doesn't return an error until the transaction is actually submitted

Reply:
Addressed by [this](https://github.com/web3/web3.js/pull/5854/commits/bf6d717b5edb77a3a26c9d0bbbccde73a993c3dc) commit

Reply:
This will not secure users 100% of times but where applicable, thats why it should not be skipped before sending tx, and its more good to check it based on defined flag. Thanks for making this change. LGTM.